server/users.ts

```diff
@@ -454,7 +457,10 @@ export class User extends Chat.MessageContext {
 	blockChallenges: boolean;
 	blockPMs: boolean | string;
 	ignoreTickets: boolean;
+	/** When the user last disconnected. */
 	lastConnected: number;
+	/** When the user last connected. */
+	lastConnectedAt: number;
```

Comment:
Why not call these `lastDisconnected`/`lastConnected`?

Reply:
I thought about this, but didn't rename it because `lastConnected` is included in `|userdetails|` and there may be bots out there that depend on it being there and it being the time a user last disconnected.

Reply:
You can change it here without changing it in `|userdetails|`.

Reply:
There are no results for bots when I search for `play.pokemonshowdown.com` and `lastConnected`, so I think it should be OK to just change the name of the property to `lastDisconnected` without needing to do anything hacky to keep it named `lastConnected` in `|userdetails|`.